base/initdefs.jl

```diff
@@ -245,8 +245,11 @@ function init_load_path()
     if haskey(ENV, "JULIA_LOAD_PATH")
         paths = parse_load_path(ENV["JULIA_LOAD_PATH"])
     else
-        paths = filter!(env -> env !== nothing,
-            String[env == "@." ? current_project() : env for env in DEFAULT_LOAD_PATH])
+        paths = String[
+            env == "@." ? current_project() : env
+            for env in DEFAULT_LOAD_PATH
+            if env !== nothing
+        ]
```

Comment:
```suggestion
        paths = String[]
        for env in DEFAULT_LOAD_PATH
            if env == "@."
                env = current_project()
                env === nothing && continue
            end
            push!(paths, env)
        end
```
since `env in DEFAULT_LOAD_PATH` never be `nothing` while `current_project()` can be?

Reply:
Very good point - I re-factored this at the last minute and turned it around entirely.